### PR TITLE
RWA-2380: Rationalise Integration test to remove duplicate tests

### PR DESCRIPTION
**Pull request by rasikasri**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceAssignTaskTest.java [hunk 1]
```diff
@@ -74,148 +67,6 @@ void setUp() {
         );
     }
 
-    @ParameterizedTest(name = "{0}")
-    @MethodSource("getGrantTypes")
-    void should_retrieve_a_task_to_assign(GrantType grantType) {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceAssignTaskTest.java:79]: Looks like we have removed this because it's IA centric. Does that mean we have the WA equivalent?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceAssignTaskTest.java [hunk 2]
```diff
@@ -74,148 +67,6 @@ void setUp() {
...
-            assigneeAccessControlResponse.getRoleAssignments(),
-            assigneePermissionsRequired
-        );
-        Assertions.assertThat(assignerTask.isPresent()).isTrue();
-        Assertions.assertThat(assignerTask.get().getTaskId()).isEqualTo(taskId);
-        Assertions.assertThat(assignerTask.get().getCaseId()).isEqualTo(caseId);
-    }
-
-    @Test
-    void should_retrieve_a_task_to_assign_challenged() {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceAssignTaskTest.java:156]: Looks like we have removed this because it's IA centric. Does that mean we have the WA equivalent?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCancelTaskTest.java [hunk 3]
```diff
@@ -71,79 +64,6 @@ void setUp() {
         );
     }
 
-    @ParameterizedTest(name = "{0}")
-    @MethodSource("getGrantTypes")
-    void should_retrieve_a_task_to_cancel(GrantType grantType) {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCancelTaskTest.java:76]: Test data setup has commonality/duplicates for getTask(...), however the permission Types differ across the various actions (This one is Cancel). Is this really a duplicate?

Looks like we have removed this because it's IA centric. Does that mean we have the WA equivalent?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCancelTaskTest.java [hunk 4]
```diff
@@ -71,79 +64,6 @@ void setUp() {
...
-
-
-        final Optional<TaskResource> task = cftQueryService.getTask(taskId, roleAssignments, permissionsRequired);
-        Assertions.assertThat(task.isPresent()).isTrue();
-        Assertions.assertThat(task.get().getTaskId()).isEqualTo(taskId);
-        Assertions.assertThat(task.get().getCaseId()).isEqualTo(caseId);
-    }
-
-    @Test
-    void should_retrieve_a_task_to_cancel_challenged() {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCancelTaskTest.java:118]: 1. Test data setup has commonality/duplicates for getTask(...), however the permission Types differ across the various actions (This one is Cancel). Is this really a duplicate?
2. Looks like we have removed this because it's IA centric. Does that mean we have the WA equivalent?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceClaimTaskTest.java [hunk 5]
```diff
@@ -71,84 +64,6 @@ void setUp() {
         );
     }
 
-    @ParameterizedTest(name = "{0}")
-    @MethodSource("getGrantTypes")
-    void should_retrieve_a_task_to_claim(GrantType grantType) {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceClaimTaskTest.java:76]: 1. Test data setup has commonality/duplicates for getTask(...), however the permission Types differ across the various actions (This one is Claim). This does not verify the scenario like assign "where it is a permission with MANAGE that will  action these tasks". Here we verify that only permission with OWN and EXECUTE can perform claim. Is this really a duplicate?
2. Looks like we have removed this because it's IA centric. Does that mean we have the WA equivalent?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceClaimTaskTest.java [hunk 6]
```diff
@@ -71,84 +64,6 @@ void setUp() {
...
-        final Optional<TaskResource> task = cftQueryService.getTask(
-            taskId, accessControlResponse.getRoleAssignments(), permissionsRequired
-        );
-        Assertions.assertThat(task.isPresent()).isTrue();
-        Assertions.assertThat(task.get().getTaskId()).isEqualTo(taskId);
-        Assertions.assertThat(task.get().getCaseId()).isEqualTo(caseId);
-    }
-
-    @Test
-    void should_retrieve_a_task_to_claim_challenged() {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceClaimTaskTest.java:121]: Same as Claim comment above.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCompleteTaskTest.java [hunk 7]
```diff
@@ -71,81 +64,6 @@ void setUp() {
         );
     }
 
-    @ParameterizedTest(name = "{0}")
-    @MethodSource("getGrantTypes")
-    void should_retrieve_a_task_to_complete(GrantType grantType) {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCompleteTaskTest.java:76]: Duplicate.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCompleteTaskTest.java [hunk 8]
```diff
@@ -71,81 +64,6 @@ void setUp() {
...
-
-
-        final Optional<TaskResource> task = cftQueryService.getTask(taskId, roleAssignments, permissionsRequired);
-        Assertions.assertThat(task.isPresent()).isTrue();
-        Assertions.assertThat(task.get().getTaskId()).isEqualTo(taskId);
-        Assertions.assertThat(task.get().getCaseId()).isEqualTo(caseId);
-    }
-
-    @Test
-    void should_retrieve_a_task_to_complete_challenged() {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCompleteTaskTest.java:119]: Duplicate

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceGetTaskTest.java [hunk 9]
```diff
@@ -154,120 +152,6 @@ void should_get_a_task_to_with_challenged_grant_type() {
         });
     }
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceGetTaskTest.java:154]: These test are focused on edge cases to ensure we elegantly return empty task. Are these permutations really duplicates or duplicated?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceUnClaimTaskTest.java [hunk 10]
```diff
@@ -71,77 +64,6 @@ void setUp() {
         );
     }
 
-    @ParameterizedTest(name = "{0}")
-    @MethodSource("getGrantTypes")
-    void should_retrieve_a_task_to_unclaim(GrantType grantType) {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceUnClaimTaskTest.java:76]: Duplicate, Assign has MANAGE verification for getTask(...) scope.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceUnClaimTaskTest.java [hunk 11]
```diff
@@ -71,77 +64,6 @@ void setUp() {
...
-        permissionsRequired.add(PermissionTypes.MANAGE);
-
-        final Optional<TaskResource> task = cftQueryService.getTask(taskId, roleAssignments, permissionsRequired);
-        Assertions.assertThat(task.isPresent()).isTrue();
-        Assertions.assertThat(task.get().getTaskId()).isEqualTo(taskId);
-        Assertions.assertThat(task.get().getCaseId()).isEqualTo(caseId);
-    }
-
-    @Test
-    void should_retrieve_a_task_to_unclaim_challenged() {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceUnClaimTaskTest.java:117]: Duplicate, Assign has MANAGE verification for getTask(...) scope.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceITTest.java [hunk 12]
```diff
@@ -119,837 +99,49 @@ void setUp() {
...
-
-            List<String> actualDates = allTasks.getTasks().stream().map(t -> t.getDueDate().format(formatter)).collect(
-                Collectors.toList());
-
-            Assertions.assertThat(actualDates.toArray()).contains(expectedDates.toArray());
-        }
-    }
-
-    @ParameterizedTest
-    @MethodSource({
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceITTest.java:168]: Do we have this covered elsewhere?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceITTest.java [hunk 13]
```diff
@@ -119,837 +99,49 @@ void setUp() {
...
-            accessControlResponse, false, scenario.granularPermission
-        );
-
-        //then
-        Assertions.assertThat(allTasks.getTasks())
-            .hasSize(scenario.expectedAmountOfTasksInResponse);
-    }
-
-    @Test
-    void handlePaginationError() {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceITTest.java:195]: Do we have this covered elsewhere?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceAssignTaskTest.java (hunk 1 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceAssignTaskTest.java:79]: Parameterized source provide only GrantType.STANDARD, GrantType.SPECIFIC. in wa/CftQueryServiceAssignTaskTest you already have two tests to cover those two grant type and some others.
should_retrieve_a_task_when_grant_type_standard
should_retrieve_a_task_when_grant_type_specific

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceAssignTaskTest.java (hunk 2 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceAssignTaskTest.java:156]: wa/CftQueryServiceAssignTaskTest 
should_retrieve_a_task_when_grant_type_challenged

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCancelTaskTest.java (hunk 4 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCancelTaskTest.java:118]: wa/CftQueryServiceCancelTaskTest
should_retrieve_a_task_when_grant_type_challenged

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceClaimTaskTest.java (hunk 5 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceClaimTaskTest.java:76]: wa/CftQueryServiceClaimTaskTest
should_retrieve_a_task_when_grant_type_standard_
should_retrieve_a_task_when_grant_type_specific_

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceClaimTaskTest.java (hunk 6 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceClaimTaskTest.java:121]: wa/CftQueryServiceClaimTaskTest
should_retrieve_a_task_when_grant_type_challenged_

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCompleteTaskTest.java (hunk 7 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCompleteTaskTest.java:76]: wa/CftQueryServiceCompleteTaskTest
should_retrieve_a_task_when_grant_type_standard
should_retrieve_a_task_when_grant_type_specific_

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCompleteTaskTest.java (hunk 8 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/ia/CftQueryServiceCompleteTaskTest.java:119]: wa/CftQueryServiceCompleteTaskTest
should_retrieve_a_task_when_grant_type_challenged
